### PR TITLE
endpoint action

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
Implementation of 'endpoint actions' for case list fields.

Spec: https://docs.google.com/document/d/1coxsFVCYgaQdxE3rgYikRSgEgistrfIUJuoW5KMiM5w/edit#heading=h.13478xeglcy4

## Technical Summary
This is a rework of:
* https://github.com/dimagi/commcare-hq/pull/33528
* https://github.com/dimagi/commcare-hq/pull/33543
* https://github.com/dimagi/commcare-hq/pull/33542

It touches the same code but implements it differently. 

This PR introduces a new Suite XML construct:

```xml
<field>
  ...
  <endpoint_action 
    endpoint_id="auto_submit_form_endpoint" 
    // Id the of the endpoint to trigger on launching this action
    background="true" 
    // Signify that the action should be launched without taking user away from the current screen 
  />
</field>
```

## Feature Flag
Clickable case list icon

### Safety story
Updates to a feature flagged feature. I did test this locally and there are also unit tests for the changes.

### Automated test coverage
Added in this PR

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -1862,17 +1862,7 @@ class DetailColumn(IndexedSchema):
...
-        # but adding in all possible reference here for safety
-        'modules[*].case_details.short.columns[*].action_form_id',
-        'modules[*].case_details.long.columns[*].action_form_id',
-        'modules[*].ref_details.short.columns[*].action_form_id',
-        'modules[*].ref_details.long.columns[*].action_form_id',
-        'modules[*].product_details.short.columns[*].action_form_id',
-        'modules[*].product_details.long.columns[*].action_form_id',
-        default="", exclude_if_none=True
-    )
+    action_endpoint_id = FormIdProperty(default="", exclude_if_none=True)
```
Comment (Robert-Costello) [inline corehq/apps/app_manager/models.py:1865]: should this be `endpoint_action_id`?

**Comment by snopoke** [comment]
@MartinRiese @shubham1g5 bump

**Review by shubham1g5 — APPROVED** [review]

**Comment by Robert-Costello** [comment]
I'm now getting this error on new builds. I'm running these and new FP changes.
`org.commcare.resources.model.UnresolvedResourceException: Invalid XML Structure(START_TAG (empty) <endpoint_action endpoint_id='favorite' background='true'>@56:66 in java.io.InputStreamReader@27a08cf8): Expected <header> Element <endpoint_action> found instead`
Could this be coming from this change?

**Comment by shubham1g5** [comment]
Probably. @snopoke We have ordering constraints with [detail field parser](https://github.com/dimagi/commcare-core/blob/formplayer/src/main/java/org/commcare/xml/DetailFieldParser.java) today, `endpoint_action` should be added after`header` and `template`.

File: corehq/apps/app_manager/suite_xml/xml_models.py [hunk 2]
```diff
@@ -825,6 +825,13 @@ class LocalizedAction(ActionMixin, TextOrDisplay):
...
+class EndpointAction(XmlObject):
+    ROOT_NAME = 'endpoint_action'
+
+    endpoint_id = StringField('@endpoint_id')
+    background = StringField('@background')
+
+
 class Field(OrderedXmlObject):
     ROOT_NAME = 'field'
     ORDER = ('style', 'header', 'template', 'sort_node')
```
Comment (Robert-Costello) [inline corehq/apps/app_manager/suite_xml/xml_models.py:837]:
```suggestion
    ORDER = ('style', 'header', 'template', 'endpoint_action', 'sort_node')
```

**Pull request merged**